### PR TITLE
P miteva/wide chars

### LINKED ISSUE #9658: Date editors should accept Japanese full-width numbers
## Is your feature request related to a problem? Please describe.
Editors should accept Japanese full-width numbers (not entered from numpad) and on blur should convert to Japanese half-width numbers. Same behavior as in the Ignite UI for [jQuery date editors](https://igniteui.com/editors/date-picker-overview)

### PR DESCRIPTION
**Pull request by PlamenaMiteva**
Closes #9658

### Additional information (check all that apply):
 - [ ] Bug fix
 - [X] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [X] All relevant tags have been applied to this PR
 - [X] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/directives/date-time-editor/date-time-editor.directive.spec.ts [hunk 1]
```diff
@@ -1005,6 +1005,27 @@ describe('IgxDateTimeEditor', () => {
...
+            it('should convert correctly full-width characters on paste', () => {
+                fixture.componentInstance.dateTimeFormat = 'dd/MM/yy';
+                fixture.detectChanges();
+                const inputDate = '０７０５２０';
+                inputElement.triggerEventHandler('focus', {});
+                fixture.detectChanges();
+                UIInteractions.simulatePaste(inputDate, inputElement, 0, 8);
+                inputElement.triggerEventHandler('blur', { target: inputElement.nativeElement });
+                fixture.detectChanges();
+                expect(inputElement.nativeElement.value).toEqual(inputDate);
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/directives/date-time-editor/date-time-editor.directive.spec.ts:1027]: Um, why would the value in the input after blur be equal to the original full-width input? Shouldn't it be parsed instead?

File: projects/igniteui-angular/src/lib/directives/mask/mask.directive.ts [hunk 2]
```diff
@@ -183,6 +185,18 @@ export class IgxMaskDirective implements OnInit, AfterViewChecked, ControlValueA
          * the end user will be unable to blur the input.
          * https://stackoverflow.com/questions/21406138/input-event-triggered-on-internet-explorer-when-placeholder-changed
          */
+        if (isComposing) {
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/directives/mask/mask.directive.ts:188]: Unfortunately, it seems `inputEvent.isComposing` is experimental still and not really available in IE11 and more importantly in Safari per https://developer.mozilla.org/en-US/docs/Web/API/InputEvent.
Considering it's basically a flag if the event is fired after compositionstart and before compositionend, perhaps it'd be more reliable if we handle that ourselves?

File: projects/igniteui-angular/src/lib/directives/date-time-editor/date-time-editor.directive.spec.ts (hunk 1 above)
Comment (damyanpetev) [reply projects/igniteui-angular/src/lib/directives/date-time-editor/date-time-editor.directive.spec.ts:1027]: Nvm, the test error sort of confirmed it for me :)

File: projects/igniteui-angular/src/lib/directives/mask/mask.directive.ts [hunk 3]
```diff
@@ -198,6 +220,34 @@ export class IgxMaskDirective implements OnInit, AfterViewChecked, ControlValueA
...
+        }
+
+        // After the compositionend event Chromium triggers input events of type 'deleteContentBackward' and
+        // we need to adjust the start and end indexes to include mask literals
+        if (event.inputType === 'deleteContentBackward' && this._key !== this.platform.KEYMAP.BACKSPACE ) {
+            let numberOfMaskLiterals = 0;
+            const literalPos = this.maskParser.getMaskLiterals(this.maskOptions.format).keys();
+            for (const index of literalPos) {
+                if (index >= this.selectionEnd && index < this._end) {
+                    numberOfMaskLiterals++;
```
Comment (jackofdiamond5) [inline projects/igniteui-angular/src/lib/directives/mask/mask.directive.ts:239]: If we have consecutive literals, this doesn't seem to work properly. Maybe we should increment `numberOfMaskliterals` based on how many consecutive literals we encounter?

maskffs

**Review by jackofdiamond5 — APPROVED** [review]

**Pull request merged**